From d6618336c1d8467a29f9221aa9ed48fab843e183 Mon Sep 17 00:00:00 2001
From: Friedemann Kleint <Friedemann.Kleint@qt.io>
Date: Fri, 10 Oct 2025 07:50:35 +0200
Subject: [PATCH 378/553] Fix QLineEdit frame color when using gradient style
 sheet and Fusion

When using a  gradient style QBrush via style sheet, an
invalid color is set on the widget palette for the Window role.

This color is used to in QFusionStylePrivate::outline() to calculate
the outline color, resulting in a black/invalid color as well for
PE_FrameLineEdit. Fall back to the widget palette in case an invalid
color is encountered.

Fixes: QTBUG-138960
Change-Id: Id1b5320476e59a743dfd0cf489032fe34e47d422
Reviewed-by: Volker Hilsheimer <volker.hilsheimer@qt.io>
(cherry picked from commit 439b758811486cbc75419df92192ef5307952ade)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/widgets/styles/qfusionstyle_p_p.h | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/widgets/styles/qfusionstyle_p_p.h b/src/widgets/styles/qfusionstyle_p_p.h
index f37ac1b72ca..dcb79f9e93c 100644
--- a/src/widgets/styles/qfusionstyle_p_p.h
+++ b/src/widgets/styles/qfusionstyle_p_p.h
@@ -71,7 +71,10 @@ public:
         }
         if (pal.window().style() == Qt::TexturePattern)
             return QColor(0, 0, 0, 160);
-        return pal.window().color().darker(140);
+        auto windowColor = pal.window().color();
+        if (!windowColor.isValid())
+            windowColor = QPalette().window().color();
+        return windowColor.darker(140);
     }
 
     QColor highlightedOutline(const QPalette &pal) const {
-- 
2.51.2

